pkg/sql/doctor/doctor.go

```diff
@@ -46,32 +62,140 @@ func NewDescGetter(rows []DescriptorTableRow) (catalog.MapDescGetter, error) {
 	return pg, nil
 }
 
+func newNamespaceMap(rows []NamespaceTableRow) namespaceReverseMap {
+	res := make(namespaceReverseMap)
+	for _, r := range rows {
+		l, ok := res[r.ID]
+		if !ok {
+			res[r.ID] = []descpb.NameInfo{r.NameInfo}
+		} else {
+			res[r.ID] = append(l, r.NameInfo)
+		}
+	}
+	return res
+}
+
 // Examine runs a suite of consistency checks over the descriptor table.
-func Examine(descTable []DescriptorTableRow, verbose bool, stdout io.Writer) (ok bool, err error) {
-	fmt.Fprintf(stdout, "Examining %d descriptors...\n", len(descTable))
-	descGetter, err := NewDescGetter(descTable)
+func Examine(
+	descTable DescriptorTable, namespaceTable NamespaceTable, verbose bool, stdout io.Writer,
+) (ok bool, err error) {
+	fmt.Fprintf(
+		stdout, "Examining %d descriptors and %d namespace entries...\n",
+		len(descTable), len(namespaceTable))
+	descGetter, err := newDescGetter(descTable)
 	if err != nil {
 		return false, err
 	}
+	nMap := newNamespaceMap(namespaceTable)
+
 	var problemsFound bool
 	for _, row := range descTable {
-		table, isTable := descGetter[descpb.ID(row.ID)].(catalog.TableDescriptor)
-		if !isTable {
-			// So far we only examine table descriptors. We may add checks for other
-			// descriptors in later versions.
+		desc, ok := descGetter[descpb.ID(row.ID)]
+		if !ok {
+			// This should never happen as ids are parsed and inserted from descTable.
+			log.Fatalf(context.Background(), "Descriptor id %d not found", row.ID)
```

Comment:
you should just pass in a context into examine, there are alot of context users here now